packages/runtime/src/runtime.ts

```diff
@@ -580,6 +621,29 @@ export class Runtime {
     }
   }
 
+  // transfer ASSET as per transaction parameters
+  transferAsset (txnParam: AssetTransferParam): void {
+    const fromAssetHolding = this.getAssetHolding(txnParam.assetID, txnParam.fromAccount.addr);
+    const toAssetHolding = this.getAssetHolding(txnParam.assetID, txnParam.toAccountAddr);
+
```

Comment:
let's add a check here for asset frozen.

Reply:
hmm, thanks. So we should also check if asset is frozen or not during opt-in?

Reply:
Added test as well.

Reply:
Thanks